examples/seq2seq/seq2seq_trainer.py

```diff
@@ -158,31 +175,35 @@ def prediction_step(
         """
         inputs = self._prepare_inputs(inputs)
 
+        if self.args.predict_with_generate and not self.args.prediction_loss_only:
+            gen_kwargs = {
+                "max_length": self.data_args.val_max_target_length,
+                "num_beams": self.data_args.eval_beams,
+            }
+            generated_tokens = model.generate(
+                inputs["input_ids"],
+                attention_mask=inputs["attention_mask"],
+                **gen_kwargs,
+            )
+            # in case the batch is shorter than max length, the output should be padded
+            if self.config.pad_token_id is not None:
```

Comment:
I would expect this case to break. `_pad_tensors_to_max_len` is needed for some sort of `Trainer`/consistent shapes reason @patil-suraj .

Reply:
Yes, `Trainer` expects all returned preds to be of same shape, which it concatenates at for every batch eval

Reply:
I don't get it -> if `config.pad_token_id` is not defined we cannot run `_pad_tensors_to_max_len`. How is this breaking anything? I am running all my experiments with no `pad_token_id` defined, so this case works.

Reply:
since `Trainer` concatenates the `preds` I assuming they should be of same length across batches. It was breaking in my last experiment when not using `_pad_tensors_to_max_len`

Reply:
I think this is fine -> see the test I added for bert2bert. Such a model does not have a `self.config.pad_token_id` defined and still works.